src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpMessageHandlerStage.cs

```diff
@@ -0,0 +1,25 @@
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    internal abstract class HttpMessageHandlerStage : HttpMessageHandler
```

Comment:
I think this class would be better named as `HttpSyncMessageHandler`

Reply:
This isn't about "Sync", though... this is creating an internal abstraction for the pipeline that SocketsHttpHandler uses, whether for sync or async.

Reply:
As Stephen says, it's used for both sync and async, the part of the name `Stage` comes from his original prototype branch, I just rolled with it.